lib/utils/feature-access.ts

```diff
@@ -0,0 +1,30 @@
+import { Limits } from "tier";
+
+function hasAccess(access: Limits["usage"], featureName: string) {
+  return !!access.find((limit) => limit.feature === `feature:${featureName}` && limit.used < limit.limit);
+}
+
+const getFeatureAccess = async () => {
+  const limits = await loadAccess();
+  const hasReports = hasAccess(limits, "reports");
+
+  return {
+    hasReports,
+  };
+};
+
+const loadAccess = async () => {
+  const response = await fetch(`/api/session`, {
+    method: "GET",
+  });
+
+  if (response.ok) {
+    return (await response.json()).access as Usage;
+  }
+
+  return [] as unknown as Usage;
```

Comment:
I'm guessing if there's a problem getting the session data we'll throw an error somewhere else in the app? Or do we need to handle it here?

Reply:
It would be somewhere else. If there is no feature access data available, this would fallback to very limited access for our current checks (reports).